packages/block-editor/src/store/private-selectors.js

```diff
@@ -480,18 +481,65 @@ export const getContentLockingParent = createSelector(
 		let current = clientId;
 		let result;
 		while ( ( current = state.blocks.parents.get( current ) ) ) {
-			if (
-				getBlockName( state, current ) === 'core/block' ||
-				getTemplateLock( state, current ) === 'contentOnly'
-			) {
+			if ( getTemplateLock( state, current ) === 'contentOnly' ) {
+				result = current;
+			}
+		}
+		return result;
+	},
+	( state ) => [
+		state.blocks.parents,
+		state.blockListSettings,
+		state.editorMode,
+		getSectionRootClientId( state ),
+	]
+);
+
+/**
+ * Retrieves the client ID of the parent section block.
+ *
+ * @param {Object} state    Global application state.
+ * @param {Object} clientId Client Id of the block.
+ *
+ * @return {?string} Client ID of the ancestor block that is content locking the block.
+ */
+export const getParentSectionBlock = createSelector(
+	( state, clientId ) => {
+		let current = clientId;
+		let result;
+		while ( ( current = state.blocks.parents.get( current ) ) ) {
+			if ( isSectionBlock( state, current ) ) {
```

Comment:
Minor/optional: There's a `getBlockParents` selector that has ascending/descending options. I think ascending would allow the loop to break early so this wouldn't have to iterate through every parent.

Reply:
> Also, I'm not sure if it's always faster, imagine a big template and you're tweaking the first child of a section within the "post content" of a complex template.

Wouldn't it always be faster with a `break`? Especially with that scenario (I think the blocks in `parents` are top to bottom). An example:
- A
  - B (section block)
    - C
      - D (another section block)
        - E
          - F (content block)

The current code would iterate from A - E and return D. With the suggestion it'd iterate from E - D and return D.

Reply:
Oh I thought it was the opposite, you're right, I prefer from E to D, it's also more accurate.

Reply:
Wait no. The current one goes from E to D. So it's the right one. Also calling `getBlockParents` would require first going through all the parents once, so an extra loop that is not needed.

Reply:
So the current code was iterating in the right order but was not breaking the loop, which I did now.

Reply:
Yeah, sorry I should've made it clearer that the `break` was the important part.

Glad we got there in the end!